### ISSUE TITLE
Option to fade out rather than immediately disappear on hide()

### ISSUE THREAD

**Issue by theduncanclark** [kiln/flourish-popup#13]
This would make it possible to include interactive elements, such as links, in the popup. Suggested by @animateddata in context of https://github.com/kiln/template-hierarchy.

**Comment by animateddata** [comment]
Before embarking on this, it's worth mapping out how this would work/help in practice.

For example, do we think the following will work:

1. Set up .on('mouseover') event on popup which calls show popup
2. Mouse moves onto element
3. Call show popup
4. Mouse moves outside element
5. Call hide popup... popup fades
6. Mouse moves on top of fading popup
7. mouseover on popup triggers, which calls show popup
8. User can now click popup

**Comment by animateddata** [comment]
And to elaborate on the original problem: we can get cases where the user has to bridge a gap between a hovered element and the popup. When the pointer leaves the hovered element, hide popup is called, unless the pointer is moved quickly enough.

It's hoped that adding a fade to the popup, the user will have plenty of time to move the pointer over the popup.

**Comment by daanlouter** [comment]
That sounds like a neat idea. Though what would happen in this case:

...
4. Mouse moves outside element
5. Call hide popup.. popup fades
_6. Mouse immediately moves onto other element_

Would the "old" popup still fade out while a new one appears? Or once your mouse moves over another element, the popup immediately moves to that new element?

**Comment by animateddata** [comment]
@daanlouter good question. I think it makes sense for the popup to immediately move (and update its content) to the new element.

**Comment by animateddata** [comment]
Would the behaviour be pretty much the same as it currently is if the user is hovering between neighbouring elements? I can't see why it would be different.

**Comment by daanlouter** [comment]
Yes, I think the behaviour would be pretty much the same. That's the reason I think it will be tricky to move your mouse to the popup to click on a link, because it's likely you'll mouse over an other region before you'll reach the popup. But I might be missing something here?

**Comment by animateddata** [comment]
Ah I see, yes, that's a good point. The behaviour's fine on the treemap, but circle packing would still have issues, because the pointer sometimes enters another element before getting to the popup... so fading might not help. (The case I was interested in was the bar layout that I'm adding to the hierarchy template. In this case, the mouse pointer would typically enter the background, rather than another element.)

Taking a step back, the actual problem is that the triangular stalk of the popup doesn't seem to react to mouse events. For example, in the treemap template, if you click on the popup it freezes. If you click in the triangular stalk, it doesn't freeze. I think this would fix more issues than fading :)
